UnleashClient/features/Feature.py

```diff
@@ -110,19 +105,53 @@ def get_variant(self, context: dict = None) -> dict:
         :param context: Context information
         :return:
         """
-        variant = DISABLED_VARIATION
-        is_feature_enabled = self.is_enabled(context)
-
-        if is_feature_enabled and self.variants is not None:
+        evaluation_result = self._get_evaluation_result(context)
+        variant = None
+        is_feature_enabled = False
+
+        if evaluation_result is not None:
+            variant = evaluation_result.variant
+            is_feature_enabled = evaluation_result.enabled
+
+        if (
+            is_feature_enabled
+            and self.variants is not None
+            and (variant is None or variant == DISABLED_VARIATION)
+        ):
             try:
                 variant = self.variants.get_variant(context)
                 variant["enabled"] = is_feature_enabled
             except Exception as variant_exception:
                 LOGGER.warning("Error selecting variant: %s", variant_exception)
 
+        if variant is None:
+            variant = copy.deepcopy(DISABLED_VARIATION)
+
         self._count_variant(cast(str, variant["name"]))
         return variant
 
+    def _get_evaluation_result(
+        self, context: dict = None, default_value: bool = False
```

Comment:
`default_value` doesn't seem to be used by this function

Reply:
it was marked as deprecated

Reply:
We should do something with it though, if we don't want to include it we can take it out, if we do then we should actually use the value. Seems weird to have a function that takes a parameter and immediately discards it

Reply:
Agreed, will remove it

Reply:
I see a rereview has been requested but I don't think this one has been done yet

Reply:
Sorry about that - must have missed it - Removed

Reply:
All good, story of my life!